src/commands/cmd_exec.nim

```diff
@@ -307,16 +443,13 @@ proc runCmdExec*(args: seq[string]) =
       initCollection()
       trace("Host collection finished.")
       let chalkOpt = doExecCollection(allOpts, ppid)
-      discard setpgid(0, 0) # Detach from the process group.
-      if isatty(0) != 0:
-        # if stdin is TTY, detach from it in child process
-        # otherwise child process will receive HUP signal
-        # on exit which is not expected
-        discard ioctl(0, TIOCNOTTY) # Detach TTY for stdin
+      detachFromParent()
       # On some platforms we don't support
       if chalkOpt.isSome():
         chalkOpt.get().collectRunTimeArtifactInfo()
       doReporting()
 
+      postExecState.doPostExec(detach = false)
```

Comment:
should detach here

will prolly end up fixing this in follow up to reduce the rebase conflicts for the future stacked PRs